source/winAPI/sessionTracking.py

```diff
@@ -106,110 +90,105 @@ class WindowsTrackedSession(enum.IntEnum):
+	"""
+	global _currentSessionState
+	log.debug("initializing session tracking")
+	_currentSessionState = _CurrentSessionState()
 
 
 def isWindowsLocked() -> bool:
 	"""
 	Checks if the Window lockscreen is active.
 	Not to be confused with the Windows sign-in screen, a secure screen.
 	"""
-	from core import _TrackNVDAInitialization
-	if not _TrackNVDAInitialization.isInitializationComplete():
+	if _currentSessionState is None:
 		# During NVDA initialization,
 		# core._initializeObjectCaches needs to cache the desktop object,
 		# regardless of lock state.
 		# Security checks may cause the desktop object to not be set if NVDA starts on the lock screen.
 		# As such, during initialization, NVDA should behave as if Windows is unlocked.
+		# sessionTracking.initialize is called when NVDA initialization is complete, which sets _currentSessionState
 		return False
-	if _isSecureDesktop():
-		# If this is the Secure Desktop,
-		# we are in secure mode and on a secure screen,
-		# e.g. on the sign-in screen.
-		# _isSecureDesktop may also return True on the lock screen before a user has signed in.
-
-		# For more information, refer to devDocs/technicalDesignOverview.md 'Logging in secure mode'
-		# and the following userGuide sections:
-		# - SystemWideParameters (information on the serviceDebug parameter)
-		# - SecureMode and SecureScreens
-		return False
-	lockStateTracked = _hasLockStateBeenTracked()
-	if lockStateTracked:
-		return WindowsTrackedSession.SESSION_LOCK in _currentSessionStates
-	else:
-		_recordLockStateTrackingFailure()  # Report error repeatedly, attention is required.
-		##
-		# For security it would be best to treat unknown as locked.
-		# However, this would make NVDA unusable.
-		# Instead, the user should be warned via UI, and allowed to mitigate the impact themselves.
-		# See usage of L{hasLockStateBeenUnknown}.
-		return False  # return False, indicating unlocked, to allow NVDA to be used
-
-
-def _setInitialWindowLockState() -> None:
-	"""
-	Ensure that session tracking state is initialized.
-	If NVDA has started on a lockScreen, it needs to be aware of this.
-	As NVDA has already registered for session tracking notifications,
-	a lock is used to prevent conflicts.
-	"""
-	with _updateSessionStateLock:
-		lockStateTracked = _hasLockStateBeenTracked()
-		if lockStateTracked:
-			log.debugWarning(
-				"Initial state already set."
-				" NVDA may have received a session change notification before initialising"
-			)
-		# Fall back to explicit query
-		try:
-			isLocked = _isWindowsLocked_checkViaSessionQuery()
-			_currentSessionStates.add(
-				WindowsTrackedSession.SESSION_LOCK
-				if isLocked
-				else WindowsTrackedSession.SESSION_UNLOCK
-			)
-		except RuntimeError as error:
-			_recordLockStateTrackingFailure(error)
+	if _currentSessionState._windowsLockStateIsUnknown:
+		from utils.security import warnSessionLockStateUnknown
+		import wx
+		wx.CallAfter(warnSessionLockStateUnknown)

```

Comment:
This might get called many times before `warnSessionLockStateUnknown` is actually called. I.E there may be many calls to `wx.CallAfter`, thus many queued calls for `warnSessionLockStateUnknown`. Please test this by setting `_windowsLockStateIsUnknown` to `True` using the python console.